### PR TITLE
feat: add newsletter worker

### PR DESCRIPTION
**Pull request by hassnian**
issue #179

needs BEEHIIV_API_KEY secret var

### CODE REVIEW

File: newsletter/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+const getResponse = (message: string) => ({ message })
+
+const app = new Hono<HonoEnv>()
+
+app.use('/subscribe', cors({ origin: allowedOrigin }))
+
+app.post('/subscribe',
+	validator('json', (value, c) => {
```
Comment (vikiival) [inline newsletter/src/index.ts:20]: Please move validator to utils

File: newsletter/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		const schema = z.object({
```
Comment (vikiival) [inline newsletter/src/index.ts:21]: Move middleware to separated function

File: newsletter/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+		return value
+	})
+	, async (c) => {
+		const { email } = c.req.valid('json')
+
+		const response = await subscribe(email, c)
+
+		if (response.status !== 201) {
+			return c.json(getResponse('Something went wrong'), 500)
```
Comment (vikiival) [inline newsletter/src/index.ts:39]: Return same status as beehiv

File: newsletter/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		}
+
+		return c.json({}, 201)
```
Comment (vikiival) [inline newsletter/src/index.ts:42]: Return 204

File: newsletter/src/utils/beehiiv.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { Context } from "hono"
+import { BEEHIIV_API_URL } from "./constants"
+import { HonoEnv } from ".."
```
Comment (vikiival) [inline newsletter/src/utils/beehiiv.ts:3]: Please move hono env somewhere else

File: newsletter/package.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "newsletter",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "deploy": "wrangler deploy",
+    "dev": "wrangler dev",
+    "start": "wrangler dev"
+  },
+  "devDependencies": {
```
Comment (vikiival) [inline newsletter/package.json:10]: Deps should be first

File: newsletter/package.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "newsletter",
+  "version": "0.0.0",
```
Comment (vikiival) [inline newsletter/package.json:3]: 0.0.1

**Review by vikiival — CHANGES_REQUESTED** [review]
Please make the code more readable. Otherwise good job

File: newsletter/src/utils/cors.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { ORIGIN } from './constants';
+
+const MATCHES: RegExp[] = [
+  /deploy-preview-[0-9]+--koda-canary.netlify.app/,
+  /deploy-preview-[0-9]+--nuxt-kodadot.netlify.app/,
+  /kodadot.xyz/,
```
Comment (roiLeo) [inline newsletter/src/utils/cors.ts:6]: missing `beta.`,`nuxt.` & `canary.`?

Comment (vikiival) [reply newsletter/src/utils/cors.ts:6]: `/kodadot.xyz/` will match all. It is the file that we are copypasting over and over ;D

Comment (roiLeo) [reply newsletter/src/utils/cors.ts:6]: yeah regex my bad ..

**Review by vikiival — APPROVED** [review]
Doable

**Comment by vikiival** [comment]
> needs BEEHIIV_API_KEY secret var

Will merge once secret will be obatined

**Comment by yangwao** [comment]
> Will merge once secret will be obatined

shared

**Comment by vikiival** [comment]
```
🌀 Creating the secret for the Worker "newsletter"
✨ Success! Uploaded secret BEEHIIV_API_KEY
```

**Pull request merged**